packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -823,25 +823,12 @@ class RenderOpacity extends RenderProxyBox {
   }
 }
 
-/// Makes its child partially transparent, driven from an [Animation].
+/// Implementation of [RenderAnimatedOpacity] and [RenderSliverAnimatedOpacity].
 ///
-/// This is a variant of [RenderOpacity] that uses an [Animation<double>] rather
-/// than a [double] to control the opacity.
-class RenderAnimatedOpacity extends RenderProxyBox {
-  /// Creates a partially transparent render object.
-  ///
-  /// The [opacity] argument must not be null.
-  RenderAnimatedOpacity({
-    @required Animation<double> opacity,
-    bool alwaysIncludeSemantics = false,
-    RenderBox child,
-  }) : assert(opacity != null),
-       assert(alwaysIncludeSemantics != null),
-       _alwaysIncludeSemantics = alwaysIncludeSemantics,
-       super(child) {
-    this.opacity = opacity;
-  }
-
+/// Use this mixin in situations where the proxying behavior
+/// of [RenderProxyBox] or [RenderProxySliver] is desired for animating opacity,
+/// but would like to use the same methods for both types of render objects.
+mixin RenderAnimatedOpacityMixin<T extends RenderObject> on RenderObjectWithChildMixin<T> {
```

Comment:
I am not sure if ```set opacity(Animation<double> value)``` is still correct in render sliver case.
It is possible the renderObject is kept alive and store in keptalive bucket. the renderObject is still 'attached' in this case. However, we don't care about its opacity in this case since it is not in the viewport.

Reply:
That's a good point. Should I check if it is visible then? Not sure how to check for that case.

Reply:
The parentdata should be able to tell. the keepalive should only happen when RenderSliver.parentdata is SliverMultiBoxAdaptorParentData. and you can check RenderSliver.parentdata.keptalive to tell if it is in keptalive bucket

Reply:
Done! :)